keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,691 @@
+
+1. Grants privileged access to the requested device:
+
+   For requests with `adminAccess: true`, the scheduler bypasses standard
+   allocation checks and allows administrators to access devices already in use.
+   This ensures privileged tasks like monitoring or diagnostics can be performed
+   without disrupting existing allocations.
+
+1. No impact on availability of claims:
+
+   The scheduler ignores claims with `adminAccess: true`, normal usage is not
+   impacted as claims in other namespaces can still be allocated using the same
+   devices that are also accessed by workloads in the admin namespace.
+
+### Workflow
+
+1. A cluster administrator labels an admin namespace with
+   `kubernetes.io/dra-admin-access`.
+
+1. Only authorized users can create `ResourceClaim` or `ResourceClaimTemplate`
+   objects with `adminAccess: true` in this admin namespace.
+
+1. Only users with access to the admin namespace can reference these
+   `ResourceClaims` or `ResourceClaimTemplates` in their pod or deployment
+   specs.
+
+1. The DRA controller processes the request and grants privileged access to the
+   requested device.
+
+## Design Details
+
+### API Changes
+
+The `adminAccess` field in `DeviceRequest` was added to `ResourceClaim` and
+`ResourceClaimTemplate` in Kubernetes v1.31:
+
+```go
+// DeviceRequest is a request for devices required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for several identical devices.
+//
+// A DeviceClassName is currently required. Clients must check that it is
+// indeed set. It's absence indicates that something changed in a way that
+// is not supported by the client yet, in which case it must refuse to
+// handle the request.
+type DeviceRequest struct {
+	...
+
+    // AdminAccess indicates that this is a claim for administrative access
+    // to the device(s). Claims with AdminAccess are expected to be used for
+    // monitoring or other management services for a device.  They ignore
+    // all ordinary claims to the device with respect to access modes and
+    // any resource allocations.
+    //
+    // This is an alpha field and requires enabling the DRAAdminAccess
+    // feature gate. Admin access is disabled if this field is unset or
+    // set to false, otherwise it is enabled.
+    //
+    // +optional
+    // +featureGate=DRAAdminAccess
+    AdminAccess *bool
+}
+```
+
+Admin access to devices is a privileged operation because it grants users access
+to devices that are in use by other users. Drivers might also remove other
+restrictions when preparing the device.
+
+In Kubernetes 1.31 (before this KEP was introduced), an example validating
+admission policy
+[was provided](https://github.com/kubernetes/kubernetes/blob/4aeaf1e99e82da8334c0d6dddd848a194cd44b4f/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml#L1-L11)
+which restricts access to this option. It is the responsibility of cluster
```

Comment:
I think "It *was formerly* the responsibility of cluster admins to ensure that such a policy *was* installed if the cluster shouldn't have allowed unrestricted access" helps keep the past tense flow clear, in that we are describing the evolution of this feature.